msal/src/Microsoft.Identity.Client/PublicClientApplication.cs

```diff
@@ -511,16 +511,43 @@ public async Task<AuthenticationResult> AcquireTokenByIntegratedWindowsAuthAsync
 
 
 #if DESKTOP
-        public async Task<AuthenticationResult> AcqjkreTokenByUsernamePassword(IEnumerable<string> scopes, System.Security.SecureString password)
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="scopes"></param>
+        /// <param name="username"></param>
+        /// <param name="password"></param>
+        /// <returns></returns>
+        public async Task<AuthenticationResult> AcquireTokenByUsernamePasswordAsync(IEnumerable<string> scopes, string username, string password)
```

Comment:
I would propose that we don't expose this overload (at least not on .net desktop where we have SecureString). The reasoning is that we should offer users a single API for achieving U/P, and we should prefer the more secure version of this API (i.e. the one at line 534 with SecureString). @henrik-me - can you think of a user scenario where they cannot use SecureString but need to use string  on net45 ?

Reply:
I can't think of a scenario right now. but there might be one... 

---
In reply to: [219455652](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1261#discussion_r219455652) [](ancestors = 219455652)